From: Varun Prakash <varun@chelsio.com>
Date: Thu, 4 Jan 2018 21:04:17 +0530
Subject: scsi: libcxgbi: use GFP_ATOMIC in cxgbi_conn_alloc_pdu()
Patch-mainline: v4.16-rc1
Git-commit: dd04b0f3b60a0144dcddf13b62392488d91bf8f6
References: bsc#1079355

For mgmt cmds ->alloc_pdu() can be called from atomic context so use
GFP_ATOMIC instead of GFP_KERNEL.

Signed-off-by: Varun Prakash <varun@chelsio.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/scsi/cxgbi/libcxgbi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/scsi/cxgbi/libcxgbi.c
+++ b/drivers/scsi/cxgbi/libcxgbi.c
@@ -1910,7 +1910,7 @@ int cxgbi_conn_alloc_pdu(struct iscsi_ta
 	if (task->sc) {
 		task->hdr = (struct iscsi_hdr *)tdata->skb->data;
 	} else {
-		task->hdr = kzalloc(SKB_TX_ISCSI_PDU_HEADER_MAX, GFP_KERNEL);
+		task->hdr = kzalloc(SKB_TX_ISCSI_PDU_HEADER_MAX, GFP_ATOMIC);
 		if (!task->hdr) {
 			__kfree_skb(tdata->skb);
 			tdata->skb = NULL;
